src/actions/actions.ts

```diff
@@ -611,6 +611,14 @@ class CommandInsertRegisterContent extends BaseCommand {
 
     if (register.text instanceof Array) {
        text = (register.text as string []).join("\n");
+    } else if (register.text instanceof RecordedState) {
+      vimState.recordedState.transformations.push({
```

Comment:
@johnfn just played with Vim again and found that running `Ctrl-R {register}` is actually performing the macro stored in `{register}`.

For example, you recored `ab<BS>c<Esc>` in register `a`, which represents `Enter Insert Mode, insert b, backspace, insert c`. Repeating this macro in Normal mode will just insert `c` and return back to Normal Mode. If you are in Insert Mode and run `<C-r> a`, what gets inserted is `ac`. The first `a` is translated to `Insert a as it's Insert Mode` and then perform all following keystrokes.